geth/account/accounts_test.go

```diff
@@ -103,9 +107,386 @@ func TestVerifyAccountPasswordWithAccountBeforeEIP55(t *testing.T) {
+	err := s.accManager.SelectAccount(s.address, s.password)
+	s.NoError(err)
+
+	testCases := []struct {
+		name                  string
+		address               string
+		password              string
+		accountKeyStoreReturn []interface{}
+		expectedError         error
+	}{
+		{
+			"success",
+			s.address,
+			s.password,
+			[]interface{}{s.keyStore, nil},
+			nil,
+		},
+		{
+			"fail_keyStore",
+			s.address,
+			s.password,
+			[]interface{}{nil, testErrKeyStore},
+			testErrKeyStore,
+		},
+		{
+			"fail_wrongAddress",
+			"wrong-address",
+			s.password,
+			[]interface{}{s.keyStore, nil},
+			ErrAddressToAccountMappingFailure,
+		},
+		{
+			"fail_wrongPassword",
+			s.address,
+			"wrong-password",
+			[]interface{}{s.keyStore, nil},
+			errors.New("cannot retrieve a valid key for a given account: could not decrypt key with given passphrase"),
+		},
+	}
+
+	for _, testCase := range testCases {
+		s.T().Run(testCase.name, func(t *testing.T) {
+			s.reinitMock()
+			s.nodeManager.EXPECT().AccountKeyStore().Return(testCase.accountKeyStoreReturn...).AnyTimes()
+			childAddr, childPubKey, err := s.accManager.CreateChildAccount(testCase.address, testCase.password)
+			if testCase.expectedError != nil {
+				s.Equal(testCase.expectedError, err)
+			} else {
+				s.NoError(err)
+				s.NotEmpty(childAddr)
+				s.NotEmpty(childPubKey)
+			}
+		})
+	}
+}
+
+func (s *ManagerTestSuite) TestSelectedAndReSelectAccount() {
+	s.reinitMock()
+
+	// Select the test account
+	s.nodeManager.EXPECT().AccountKeyStore().Return(s.keyStore, nil).AnyTimes()
+	s.nodeManager.EXPECT().WhisperService().Return(s.shh, nil).AnyTimes()
+	err := s.accManager.SelectAccount(s.address, s.password)
+	s.NoError(err)
+
+	s.T().Run("success", func(t *testing.T) {
+		acc, err := s.accManager.SelectedAccount()
+		s.NoError(err)
+		s.NotNil(acc)
+
+		err = s.accManager.ReSelectAccount()
+		s.NoError(err)
+	})
+
+	s.T().Run("ReSelect_fail_whisper", func(t *testing.T) {
+		s.reinitMock()
+		s.nodeManager.EXPECT().WhisperService().Return(nil, testErrWhisper).AnyTimes()
+		err = s.accManager.ReSelectAccount()
+		s.Equal(testErrWhisper, err)
+	})
+
+	s.accManager.selectedAccount = nil
+	s.reinitMock()
+	s.nodeManager.EXPECT().AccountKeyStore().Return(s.keyStore, nil).AnyTimes()
+	s.nodeManager.EXPECT().WhisperService().Return(s.shh, nil).AnyTimes()
+
+	s.T().Run("Selected_fail_noAccount", func(t *testing.T) {
+		_, err := s.accManager.SelectedAccount()
+		s.Equal(ErrNoAccountSelected, err)
+	})
+
+	s.T().Run("ReSelect_success_noAccount", func(t *testing.T) {
+		err = s.accManager.ReSelectAccount()
+		s.NoError(err)
+	})
+}
+
+func (s *ManagerTestSuite) TestLogout() {
+	s.reinitMock()
+
+	s.nodeManager.EXPECT().WhisperService().Return(s.shh, nil)
+	err := s.accManager.Logout()
+	s.NoError(err)
+
+	s.nodeManager.EXPECT().WhisperService().Return(nil, testErrWhisper)
+	err = s.accManager.Logout()
+	s.Equal(testErrWhisper, err)
+}
+
+func (s *ManagerTestSuite) TestAccounts() {
```

Comment:
`TestAccounts` actually doesn't say much about the test. Could you rename it accoording to test case below?

Reply:
Every test function is named using the common naming convention such as `TestFunctionName` and function name is only `Accounts` in this case. I hope it makes sense.

Reply:
Hehe, yeah, typical trouble due to the naming convention. That's why I personally for my code also like little comments for each test.

But here for me it's okay.

Reply:
Hmm. Let's just leave a short comment to make it more obvious.